blocks/library/embed/editor.scss

```diff
@@ -1,7 +1,15 @@
 .editor-block-list__block[data-type="core/embed"] {
 	&[data-align="left"],
 	&[data-align="right"] {
-		max-width: 100%;
+		float: none;
+	}
+
+	&[data-align="left"] .wp-block-embed {
+		float: left;
+	}
+
+	&[data-align="right"] .wp-block-embed {
+		float: right;
```

Comment:
Can you explain why the embed block needs specific handling of floats? Trying to understand to see if we can come up with something generic to avoid all styles in individual blocks

Reply:
I forked this branch and added this commit https://github.com/WordPress/gutenberg/commit/003c8da7c9f742c30f56c313c5a8e44fe122a2d6 to avoid any special casing in blocks. Why can't this work?

Reply:
There are still some specific styles for the button block because of the URL input but that's fine. So now the generic use-case when aligning left/right assigns a max-width unless the data-resized is true which indicates the width should be automatic because of the content of the block.

What do you think?

Reply:
I like that. Sounds good to me for sure. I'll see if I can port your changes, otherwise feel free to push to the branch.